ui/modal/modalRepost/view.jsx

```diff
@@ -0,0 +1,199 @@
+  doClearRepostError: () => void,
+  doRepost: StreamRepostOptions => Promise<*>,
+  title: string,
+  claim: ?StreamClaim,
+  balance: number,
+  channels: ?Array<ChannelClaim>,
+  myClaims: ?Array<StreamClaim>,
+  doFetchClaimListMine: () => void,
+  error: ?string,
+  reposting: boolean,
+};
+
+function ModalRepost(props: Props) {
+  const {
+    doHideModal,
+    doToast,
+    doClearRepostError,
+    doRepost,
+    title,
+    claim,
+    balance,
+    channels,
+    error,
+    reposting,
+    myClaims,
+    doFetchClaimListMine,
+  } = props;
+  const defaultName = claim && `${claim.name}-repost`;
+  const contentClaimId = claim && claim.claim_id;
+  const [repostChannel, setRepostChannel] = usePersistedState('repost-channel');
+  const [repostBid, setRepostBid] = usePersistedState('repost-bid', 0.1);
+  const [showAdvanced, setShowAdvanced] = React.useState();
+  const [repostName, setRepostName] = React.useState(defaultName);
+
+  let repostBidError;
+  if (repostBid === 0) {
+    repostBidError = __('Deposit cannot be 0');
+  } else if (balance === repostBid) {
+    repostBidError = __('Please decrease your deposit to account for transaction fees');
+  } else if (balance < repostBid) {
+    repostBidError = __('Deposit cannot be higher than your balance');
+  } else if (repostBid < MINIMUM_PUBLISH_BID) {
+    repostBidError = __('Your deposit must be higher');
+  }
+
+  let repostNameError;
+  if (!repostName) {
+    repostNameError = __('A name is required');
+  } else if (!isNameValid(repostName, false)) {
+    repostNameError = INVALID_NAME_ERROR;
+  } else if (
+    channels &&
+    channels.find(claim => claim.name === repostChannel) &&
+    myClaims &&
+    myClaims.find(claim => claim.name === repostName)
+  ) {
+    repostNameError = __('You already have a claim with this name.');
+  }
+
+  const channelStrings = channels && channels.map(channel => channel.permanent_url).join(',');
+  React.useEffect(() => {
+    if (!repostChannel && channelStrings) {
+      const channels = channelStrings.split(',');
+      const newChannelUrl = channels[0];
+      const { claimName } = parseURI(newChannelUrl);
+      setRepostChannel(claimName);
+    }
+  }, [channelStrings]);
+
+  const myClaimsString = myClaims && myClaims.map(channel => channel.permanent_url).join(',');
+  React.useEffect(() => {
+    if (myClaimsString === '') {
+      doFetchClaimListMine();
+    }
+  }, [myClaimsString, doFetchClaimListMine]);
+
+  function handleSubmit() {
+    const channelToRepostTo = channels && channels.find(channel => channel.name === repostChannel);
+    if (channelToRepostTo && repostName && repostBid && repostChannel && contentClaimId) {
+      doRepost({
+        name: repostName,
+        bid: creditsToString(repostBid),
+        channel_id: channelToRepostTo.claim_id,
+        claim_id: contentClaimId,
+      }).then(() => {
+        doHideModal();
+        doToast({ message: __('Woohoo! Sucessfully reposted this claim.') });
```

Comment:
can link to your channel page

Reply:
It won't be displayed on the channel page until it's confirmed